src/controllers/user.controller.ts

```diff
@@ -189,3 +189,47 @@ export const patchMarkUserAdminHandler = async (
 		});
 	}
 };
+
+/**
+ * This controller will unscribe user marketing mail.
+ *
+ * @param req request
+ * @param res response
+ *
+ * @author arbaz23
+ */
+export const updateUnsubcribeUser = async (
+	req: Request<PatchUserSchema["params"]>,
+	res: Response,
+) => {
+	const { id } = req.params;
+
+	try {
+		// check if receiveMarketingEmails is already false
+		const UserWithreceiveMarketingEmailsTrue = await findUserByIdService(id || "");
+
+		if (!UserWithreceiveMarketingEmailsTrue.receiveMarketingEmails) {
+			return res.status(StatusCodes.CONFLICT).json({
+				error: "User is already unsubcribe",
+			});
+		}
+		const updateUser = await updateUserByIdService(id, {receiveMarketingEmails:
+			false});
+
+		if (!updateUser) {
```

Comment:
This check isn't required. Use the middleware to take care of it https://github.com/MultiEmail/backend/blob/e1adbb9e341189cde8cabfc8e17791650a3338b9/src/middleware/deserializeUser.middleware.ts#L26